bitsandbytes/autograd/_functions.py

```diff
@@ -1,20 +1,14 @@
 from dataclasses import dataclass
-from functools import reduce  # Required in Python 3
-import operator
+from math import prod
 from typing import Callable, Optional, Tuple
 import warnings
 from warnings import warn
 
 import torch
+from typing_extensions import deprecated
```

Comment:
I'd avoid a new dependency just for this, but it's very likely `typing_extensions` will be in the venvs for users anyway...

Reply:
This is a dependency of `torch` so it seems reasonable to me.